cifar10.py

```diff
@@ -29,58 +43,147 @@
+    print(X_train.shape[0], 'train samples')
+    print(X_test.shape[0], 'test samples')
+
+    # Convert class vectors to binary class matrices.
+    Y_train = np_utils.to_categorical(y_train, nb_classes)
+    Y_test = np_utils.to_categorical(y_test, nb_classes)
+
+    X_train = X_train.astype('float32')
+    X_test = X_test.astype('float32')
+    # TODO switch to ResnetBuilder.build calls, permute parameters in 3x3 grid, generate plots for final assignment, save models?
+    # use https://github.com/fchollet/keras/blob/master/examples/lstm_benchmark.py
+    model = resnet.ResnetBuilder.build_resnet_18((img_channels, img_rows, img_cols), nb_classes)
+
+    # subtract mean and normalize
+    mean_image = np.mean(X_train, axis=0)
+    X_train -= mean_image
+    X_test -= mean_image
+    X_train /= 128.
+    X_test /= 128.
+
+    model = resnet.ResnetBuilder.build_resnet_18((img_channels, img_rows, img_cols), nb_classes)
+    model.compile(loss='categorical_crossentropy',
+                  optimizer='adam',
+                  metrics=['accuracy'])
+
+    if not data_augmentation:
+        print('Not using data augmentation.')
+        model.fit(X_train, Y_train,
+                  batch_size=batch_size,
+                  nb_epoch=nb_epoch,
+                  validation_data=(X_test, Y_test),
+                  shuffle=True,
+                  callbacks=[lr_reducer, early_stopper, csv_logger])
+    else:
+        print('Using real-time data augmentation.')
+        # This will do preprocessing and realtime data augmentation:
+        datagen = ImageDataGenerator(
+            featurewise_center=False,  # set input mean to 0 over the dataset
+            samplewise_center=False,  # set each sample mean to 0
+            featurewise_std_normalization=False,  # divide inputs by std of the dataset
+            samplewise_std_normalization=False,  # divide each input by its std
+            zca_whitening=False,  # apply ZCA whitening
+            rotation_range=0,  # randomly rotate images in the range (degrees, 0 to 180)
+            width_shift_range=0.1,  # randomly shift images horizontally (fraction of total width)
+            height_shift_range=0.1,  # randomly shift images vertically (fraction of total height)
+            horizontal_flip=True,  # randomly flip images
+            vertical_flip=False)  # randomly flip images
+
+        # Compute quantities required for featurewise normalization
+        # (std, mean, and principal components if ZCA whitening is applied).
+        datagen.fit(X_train)
+        if not os.path.exists(out_dir):
+            os.makedirs(out_dir)
+        
+        csv = CSVLogger(out_dir+dirname+'.csv', separator=',', append=True)
+        model_checkpoint = ModelCheckpoint(out_dir+'weights.hdf5', monitor='val_loss', verbose=0, save_best_only=True, save_weights_only=False, mode='auto')
+        callbacks=[lr_reducer, early_stopper, csv]
+        
+        if K.image_dim_ordering() == 'tf':
```

Comment:
image dim ordering doesn't imply tensorflow backend. Is there a way to check 'backend' instead?